Request/ParamConverter/DoctrineParamConverter.php

```diff
@@ -229,12 +267,33 @@ public function supports(ParamConverter $configuration)
 
     private function getOptions(ParamConverter $configuration)
     {
-        return array_replace(array(
+        $defaultValues = array(
             'entity_manager' => null,
             'exclude' => array(),
             'mapping' => array(),
             'strip_null' => false,
-        ), $configuration->getOptions());
+            'expr' => null,
+            'id' => null,
+            'repository_method' => null,
+            'map_method_signature' => false,
+        );
+
+        $passedOptions = $configuration->getOptions();
+
+        if (isset($passedOptions['repository_method'])) {
+            @trigger_error('The repository_method option of @ParamConverter is deprecated and will be removed in 5.0. Use the expr option or @Entity.', E_USER_DEPRECATED);
+        }
+
+        if (isset($passedOptions['map_method_signature'])) {
+            @trigger_error('The map_method_signature option of @ParamConverter is deprecated and will be removed in 5.0. Use the expr option or @Entity.', E_USER_DEPRECATED);
+        }
```

Comment:
We could also consider deprecating `exclude` and maybe `mapping`, though to replace `mapping` with an `expr` is still kind of ugly:

Before:
```php
@Entity("comment", options={"mapping": {"comment_slug": "slug"}})
```

After:
```php
 @Entity("comment", expr="repository.findOneBy({'comment_slug': slug})")
```